src/WebJobs.Script.WebHost/Middleware/CustomHeadersMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Middleware;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class CustomHeadersMiddleware : IJobHostHttpMiddleware
+    {
+        private RequestDelegate _invoke;
+
+        public CustomHeadersMiddleware(IOptions<ScriptJobHostOptions> hostOptions)
+        {
+            RequestDelegate contextNext = async context =>
```

Comment:
I'm assuming this pattern is in place here because of what was done in the HSTS middleware, but this middleware doesn't have the same requirements, so you should be able to simplify this and make it a bit more efficient.

Something as simple as this should just work:

``` csharp
 public class CustomHeadersMiddleware : IJobHostHttpMiddleware
    {
        private readonly ScriptJobHostOptions _hostOptions;

        public CustomHeadersMiddleware(IOptions<ScriptJobHostOptions> hostOptions)
        {
            _hostOptions = hostOptions.Value;
        }

        public async Task Invoke(HttpContext context, RequestDelegate next)
        {
            await next(context);

            foreach (var header in _hostOptions.CustomHeaders)
            {
                context.Response.Headers[header.Key] = header.Value;
            }
        }
    }
```

Reply:
Can do. More for my own curiosity than anything else, and to better understand this part of the codebase, what were the requirements for the HSTS middleware that necessitated the pattern Naren used?